msrest/serialization.py

```diff
@@ -1086,7 +1086,7 @@ def rest_key_extractor(attr, attr_desc, data):
     key = attr_desc['key']
     working_data = data
 
-    while '.' in key:
+    while '.' in key and working_data:
```

Comment:
That's not enough, when it's None you will end the loop and still fail line 1098

Reply:
Hmm... Nice catch, but actually in that case `data` is `None` which is against our purpose of calling the function. :)

Reply:
Nice catch. I only looked at the code at L1113~L1114, but didn't notice L1098. I thought they were the same. :P